src/app/core/authentication/auth.service.ts

```diff
@@ -26,7 +26,8 @@ export class AuthService {
     );
 
     return merge(token$, refresh$).pipe(
-      switchMap(() => (this.check() ? this.assignUser() : this.assignGuest()))
+      switchMap(() => (this.check() ? this.assignUser() : this.assignGuest())),
+      map(() => this.check())
```

Comment:
if the auth onchange return user information, there are two ways to get user information (onChange and user)
the developer could be confused, so I want to make get user information only using user method.

maybe I can use APP_INITIALIZER to initialize auth service, then it only call AuthService.user() to detect change at StartupService

Reply:
But if onChange return the boolean, there are also two ways to get authenticated boolean (onChange and check). Maybe the user information is more important than boolean.

Reply:
I want to make some changes, but this pr be merged XD

Reply:
because AuthService.user() will return BehaviorSubject, the developer can get user information whenever call subscribe but AuthService.onChange does not.

And I send another pr. AuthService uses APP_INITIALIZER to initialize AuthService and remove AuthService.onChange, because the developer doesn't need to initialize AuthService by himself, and onChange can be removed.

I think AuthService is easier to use when AuthService has a less public method